src/Builder/ResponseBuilder.php

```diff
@@ -135,8 +135,8 @@ public function addHeader($headerLine)
                 sprintf('"%s" is not a valid HTTP header line', $headerLine)
             );
         }
-        $name = trim(urldecode($parts[0]));
-        $value = trim(urldecode($parts[1]));
+        $name = trim($parts[0]);
+        $value = trim($parts[1]);
```

Comment:
It seems to me the urldecode is not necessary, only ascii characters are allowed.